### PR TITLE
6 get /api/reviews/review id/comments

### PR DESCRIPTION
**Pull request by rovercode1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -87,68 +147,65 @@ describe('api',()=>{
...
-    })
-      
-    })
-  })
-})
+        .get("/api/reviews/3/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const comments = body.comments;
+          // comments should be served with the most recent comments first
```
Comment (HannahBurks) [inline __tests__/app.test.js:182]: Lets remove this comment :)

File: models/comment-models.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const format = require("pg-format");
+
+exports.selectComments = (reviewId) => {
+  let queryStr = 'SELECT * FROM comments'
+  let queryParam = []
+
+
+  const regEx = /^\d+$/gm;
+
+  if (regEx.test(parseInt(reviewId)) === false) {
```
Comment (HannahBurks) [inline models/comment-models.js]: If it is an invalid ID (such as a string) PSQL should automatically throw and error due to it breaking the sql type - you can then instead handle this error in your error handler by checking the psql error code :) We won't need to deal with it here, only the error from an id that does not exists such as 10000) - then we can do a promise .reject if we get no data back

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Great work so far, just an error handling change as we want to utilising our PSQL errors which are automatically thrown. Do a .only on the 400 test and console log err in the error handler file - you should see the psql error if you try to put a string in as ID

File: __tests__/app.test.js [hunk 3]
```diff
@@ -87,68 +147,64 @@ describe('api',()=>{
...
+            expect(typeof comment.votes).toBe("number");
+            expect(typeof comment.created_at).toBe("string");
+            expect(typeof comment.author).toBe("string");
+            expect(typeof comment.body).toBe("string");
+            expect(typeof comment.review_id).toBe("number");
+          });
+        });
+      ///
+    });
+
```
Comment (sirKevlar) [inline __tests__/app.test.js:175]: We need one more happy path test here... There is a review in your data which has no associated comments. In this case, we would expect 200 and an empty array would be returned

**Pull request merged**